python/tank/descriptor/io_descriptor/github_release.py

```diff
@@ -0,0 +1,269 @@
+        """
+        Returns a descriptor object that represents the latest version
+        that is locally available in the bundle cache search path.
+
+        :param constraint_pattern: If this is specified, the query will be constrained
+               by the given pattern. Version patterns are on the following forms:
+
+                - v0.1.2, v0.12.3.2, v0.1.3beta - a specific version
+                - v0.12.x - get the highest v0.12 version
+                - v1.x.x - get the highest v1 version
+
+        :returns: instance deriving from IODescriptorBase or None if not found
+        """
+        all_versions = self._get_locally_cached_versions()
+        version_numbers = all_versions.keys()
+
+        if not version_numbers:
+            return None
+
+        version_to_use = self._find_latest_tag_by_pattern(version_numbers, constraint_pattern)
+        if version_to_use is None:
+            return None
+
+        # make a descriptor dict
+        descriptor_dict = {
+            "organization": self._organization,
+            "repository": self.get_system_name(),
+            "version": version_to_use,
+            "type": "github_release",
+        }
+
+        # and return a descriptor instance
+        desc = IODescriptorGithubRelease(descriptor_dict, self._sg_connection, self._bundle_type)
+        desc.set_cache_roots(self._bundle_cache_root, self._fallback_roots)
+
+        log.debug("Latest cached version resolved to %r" % desc)
+        return desc
+
+    def has_remote_access(self):
+        """
+        Probes if the current descriptor is able to handle
+        remote requests. If this method returns, true, operations
+        such as :meth:`download_local` and :meth:`get_latest_version`
+        can be expected to succeed.
+
+        :return: True if a remote is accessible, false if not.
+        """
+        # check if we can make api request for the specified repo
+        can_connect = True
+        url = "https://api.github.com/repos/{organization}/{system_name}"
+        url = url.format(organization=self._organization, system_name=self.get_system_name())
+        if self._sg_connection.config.proxy_handler:
+            # Grab proxy server settings from the shotgun API
+            opener = urllib2.build_opener(self._sg_connection.config.proxy_handler)
+            urllib2.install_opener(opener)
+        try:
+            log.debug("%r: Probing if a connection to Github can be established..." % self)
+            # ensure we get response code 200
+            response_code = urllib2.urlopen(url).getcode()
+            # Unfortunately, to prevent probing private repos, GH API also gives a 404 response
+            # for private repos accessed without a token, so there's no way to helpfully warn the
+            # user if they try to download from a private repo.
+            can_connect = response_code == 200
+            # @todo Should we also deal with redirects here?
```

Comment:
In the case that a repository is renamed, GH API can give a 300 response. 
 https://developer.github.com/v3/#http-redirects
Right now this would result in `has_remote_access` returning False.

Reply:
Have you looked into [urllib2.HTTPRedirectHandler](https://docs.python.org/2/library/urllib2.html#urllib2.HTTPRedirectHandler). It will handle redirects for you transparently.

Reply:
great! I suggest we add that as a comment to the code for now.

Reply:
Added to the comment, but didn't change the code.